### PR TITLE
Fix KeyVault core reference to runtime and add constructor with httpclient

### PR DESCRIPTION
**Pull request by pomortaz**
- When key vault users are getting KeyVault.Core nuget package, they get an error related to GAC and System.Runtime. The dependency is unnecessary for net451.
- There is an ask by developers regarding exposing HttpClient in the constructor.

### CODE REVIEW

**Comment by pomortaz** [comment]
@herveyw please review

**Comment by herveyw** [comment]
@pomortaz Looks fine, I think you'll need to bump the KeyVault package version if you want the new NuGet description and release notes to be published.

**Comment by pomortaz** [comment]
@herveyw we are going to bump the SDK package version soon with new API version. I'll let it handle the new package creation.

**Comment by kunalcha** [comment]
:shipit:

**Comment by markcowl** [comment]
@pomortaz Please fix merge conflicts

**Comment by pomortaz** [comment]
@markcowl can you please pull the changes?

File: src/KeyVault/Microsoft.Azure.KeyVault.Core/project.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "2.0.2-preview",
+  "version": "2.0.3-preview",
```
Comment (hovsepm) [inline src/KeyVault/Microsoft.Azure.KeyVault.Core/project.json]: Please also update AssemblyFileVersion in AsseblyInfo.cs file

**Review by hovsepm — CHANGES_REQUESTED** [review]

File: src/KeyVault/Microsoft.Azure.KeyVault.Core/project.json (hunk 1 above)
Comment (pomortaz) [inline src/KeyVault/Microsoft.Azure.KeyVault.Core/project.json]: Thanks. Added the AssemblyFileVersion.

**Review by hovsepm — APPROVED** [review]

**Pull request merged**